doc/md/guides/testing/testing-migrations.mdx

```diff
@@ -0,0 +1,276 @@
+    ref_columns = [table.users.column.id]
+  }
+  foreign_key "liked_post_fk" {
+    columns = [column.post_id]
+    ref_columns = [table.posts.column.id]
+  }
+}
+```
+
+### Generating a Migration
+
+To generate our first migration, we will run the following command:
+
+```bash
+atlas migrate diff --env dev
+```
+
+We should see a new `migrations` directory created with the following files:
+
+<Tabs>
+<TabItem value="20240730073333.sql" label="20240730073333.sql" default>
+
+```sql title="20240730073333.sql"
+-- Create "users" table
+CREATE TABLE "public"."users" ("id" integer NOT NULL, "email" character varying(255) NOT NULL, PRIMARY KEY ("id"));
+-- Create "posts" table
+CREATE TABLE "public"."posts" ("id" integer NOT NULL, "title" character varying(100) NOT NULL, "user_id" integer NOT NULL, PRIMARY KEY ("id"), CONSTRAINT "authors_fk" FOREIGN KEY ("user_id") REFERENCES "public"."users" ("id") ON UPDATE NO ACTION ON DELETE NO ACTION);
+-- Create "likes" table
+CREATE TABLE "public"."likes" ("id" integer NOT NULL, "user_id" integer NOT NULL, "post_id" integer NOT NULL, PRIMARY KEY ("id"), CONSTRAINT "liked_by_fk" FOREIGN KEY ("user_id") REFERENCES "public"."users" ("id") ON UPDATE NO ACTION ON DELETE NO ACTION, CONSTRAINT "liked_post_fk" FOREIGN KEY ("post_id") REFERENCES "public"."posts" ("id") ON UPDATE NO ACTION ON DELETE NO ACTION);
+```
+
+</TabItem>
+<TabItem value="atlas.sum" label="atlas.sum">
+
+```sum title="atlas.sum"
+h1:gUOA9cOgCheiTimm/l1DTlIyUGW1vRY9pKtjXlwp5sY=
+20240730073333.sql h1:PpPJDqDZcNZTFUl26jYUvPMonpknEMiwuevh7FkGoPA=
+```
+
+</TabItem>
+</Tabs>
+
+### Expanding Business Logic
+
+Great! Now that we have a basic schema and migration directory, let's add some business logic before
+we begin testing.
+
+We will add a view `post_likes_count` that will count the number of likes each post gets:
+
+```hcl title="schema.hcl"
+view "post_likes_count" {
+  schema = schema.public
+  column "post_id" {
+    null = false
+    type = integer
+  }
+  column "title" {
+    null = false
+    type = text
+  }
+  column "likes" {
+    null = false
+    type = bigint
+  }
+  as         = <<-SQL
+   SELECT p.id AS post_id,
+      p.title,
+      count(l.id) AS likes
+     FROM (public.posts p
+       LEFT JOIN public.likes l ON ((p.id = l.post_id)))
+    GROUP BY p.id, p.title;
+  SQL
+  depends_on = [table.likes, table.posts]
+}
+```
+
+Run the `migrate diff` command once more to generate another migration:
+
+```bash
+atlas migrate diff --env dev
+```
+
+The following migration should be created:
+
+```sql title="20240730073842.sql"
+-- Create "post_likes_count" view
+CREATE VIEW "public"."post_likes_count" ("post_id", "title", "likes") AS SELECT p.id AS post_id,
+    p.title,
+    count(l.id) AS likes
+   FROM (public.posts p
+     LEFT JOIN public.likes l ON ((p.id = l.post_id)))
+  GROUP BY p.id, p.title;
+```
+
+## Testing Migrations
+
+Now that we have our project setup, we are ready to begin writing tests!
```

Comment:
the example isn't the best we can provide. testing a view should be done using `schema test`. in your example there should be a re-writing of data. e.g, to improve performance we want to add the timestamp of the most recent post as `latest_post_ts` to the user. we add a trigger that updates this - for future data, but for existing data we must populate the table with seed data. so we generate the migration, then edit it with some seeding logic, the migrate hash, then run the test.